### DISCUSSION TITLE
Cannot pipe into a command with quotes on Windows

### DISCUSSION THREAD

**Discussion by Visic** [helix-editor/helix#7752] [Troubleshooting]
### Summary

I tried to replace some text by piping it through sed, but it looks like the pipe command is escaping quotes which makes the command invalid.

![image](https://user-images.githubusercontent.com/2312662/229633344-d32a4c12-7ea1-4f59-b46d-4c23b354784a.png)

![image](https://user-images.githubusercontent.com/2312662/229634971-0102440b-663f-4d77-bf6d-f97dfa085278.png)

In cmd:
![image](https://user-images.githubusercontent.com/2312662/229633565-c35a4a40-0a9c-4d8b-a12b-3dae0390c67f.png)

An illustrative example of the quote escaping using echo:
![image](https://user-images.githubusercontent.com/2312662/229635141-d76bae1e-6061-48aa-8b11-7a5b4dd21926.png)

My use case was easy enough to work around for now outside of helix, but I wanted to leave a report in case it isn't intended to be this way.

### Platform

Windows 11

### Terminal Emulator

winterm

### Helix Version

23.03

**Comment by kirawi** [comment]
Probably like https://matrix.to/#/!zMuVRxoqjyxyjSEBXc:matrix.org/$kJwFOYbl9yb6fK7YdYcqHPIUt2ADz2MmNcot06xRAcQ?via=matrix.org&via=mozilla.org&via=envs.net

> I ran into this before. Pretty sure this is not a new regression. The problem here is that there are multiple people escaping. Helix escapes everything in command mode but so does your shell so it's pretty hard to get everyone to "shutup and accept my text.
> 
> For example try :sh echo \\\\\\"foo\\\\\\" This will print "foo".
> This works as follows:
> * helix first escapes this to echo \\"foo\\"
> * Your shell then escapes that to echo "foo"

**Comment by chtenb** [comment]
Is it documented somewhere how the escaping rules work for `:sh` and cousins? I'm trying to configure a language formatter (for removing trailing wihtespace), but I can't seem to get the arguments to `sed` right (on windows). Whatever I try, the pattern doesn't get through correctly

**Reply by kirawi** [reply]
It's not, and we want to remove escaping on shell commands (I'll create an issue for it)
